### PR TITLE
added helper

### LINKED ISSUE #45: Create About or Help Dialog
We need help dialog or at least about dialog to point the user what's the plugin about and how to use it. No need to have a help in the wizard, since it should be self-explanatory.

### PR DESCRIPTION
**Pull request by meomancer**
Added helper fix #45 :
- [x] Main helper
- [x] Download Project helper
- [x] Create Project helper
- [ ] Update Project helper
- [x] Contact helper
screenshot from 2017-01-02 14 03 44

### CODE REVIEW

**Comment by timlinux** [comment]
For a simple help system you may want to look at the messaging classes from InaSAFE (@ismailsunni we could hive those out to a separate package / repo). They have the advantage that you do not need separate workflows for translating project documentation and UI stuff. See e.g. 

https://github.com/inasafe/inasafe/blob/develop/safe/gui/tools/help/function_options_help.py

**Comment by ismailsunni** [comment]
Hi @timlinux 
Yeah, I think we can do it. But the messaging classes are not completely separated from the other code. It imports `tr`, `get_unicode`, and `resource_path`. Perhaps we can move them out to a separate package.  I checked, those three helper methods are independent from the rest of the code. Then we can have `qgis_plugin_utilities` repository that people can re-use it also.

**Comment by meomancer** [comment]
screenshot from 2017-01-03 13 12 53

Using message class

**Comment by ismailsunni** [comment]
Looks very nice to me

**Comment by Gustry** [comment]
>Then we can have qgis_plugin_utilities repository that people can re-use it also.

They are many files that we need to copy and paste when we work on another plugin : messaging system, datastore (folder, geopackage, postgis ...), helpers to load a ressource, helpers to load a vector/raster layer for tests, debug helpers ... It could be very nice if we can refactor these.

File: cadasta/gui/tools/helper/content/create_project_help.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    :returns: A message object without brand element.
+    :rtype: safe.messaging.message.Message
+    """
+
+    message = m.Message()
+
+    message.add(m.Paragraph(tr(
+        'Create Project is one of main wizard that will help you to create'
+        'project in qgis and update it to Cadasta with easy steps.'
+        '<b>This wizard</b>')))
```
Comment (Gustry) [inline cadasta/gui/tools/helper/content/create_project_help.py]: I'm not a messaging expert, but I think we shouldn't use HTML in the message itself. We can export the message as markdown, json ...
There is also some html tags below in the code.

File: cadasta/gui/tools/helper/content/create_project_help.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    bullets.add(m.Text(
+        m.ImportantText(tr('Project Name')),
+        tr('- it is name for new project')
+    ))
+    bullets.add(m.Text(
+        m.ImportantText(tr('Project Url')),
+        tr('- it is url for new project')
+    ))
+    bullets.add(m.Text(
+        m.ImportantText(tr('QGis Layer to upload')),
```
Comment (Gustry) [inline cadasta/gui/tools/helper/content/create_project_help.py]: The name is `QGIS`. But I think we can just write `Layer to upload`.

File: cadasta/gui/tools/helper/content/create_project_help.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        tr('- select qgis layer that created in qgis.')
```
Comment (Gustry) [inline cadasta/gui/tools/helper/content/create_project_help.py]: Same here about QGIS. I think `Select the layer that was created` ?

File: cadasta/gui/tools/helper/content/download_project_help.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+    :returns: A message object without brand element.
+    :rtype: safe.messaging.message.Message
+    """
+
+    message = m.Message()
+
+    message.add(m.Paragraph(tr(
+        'Download Project is one of main wizard that will help you to get '
+        'data from Cadasta. With easy step, you will get data from cadasta, '
```
Comment (Gustry) [inline cadasta/gui/tools/helper/content/download_project_help.py]: cadasta -> Cadasta
QGis -> QGIS

File: cadasta/gui/tools/wizard/step_project_update02.py [hunk 5]
```diff
@@ -92,7 +92,7 @@ def set_widgets(self):
 
         # From server
         for index, contact in enumerate(project_contacts):
-
+            print index
```
Comment (Gustry) [inline cadasta/gui/tools/wizard/step_project_update02.py]: Remove the print

**Comment by timlinux** [comment]
Looking nice @meomancer !

**Comment by ismailsunni** [comment]
> They are many files that we need to copy and paste when we work on another plugin : messaging system, datastore (folder, geopackage, postgis ...), helpers to load a ressource, helpers to load a vector/raster layer for tests, debug helpers ... It could be very nice if we can refactor these.

Yeah, totally agree with this.

**Comment by ismailsunni** [comment]
Hi @meomancer if you want to merge this PR, please go. The PR is getting bigger. You can make another PR for the Update Project Helper.

**Pull request merged**